### PR TITLE
Add Blob Model Constructor

### PR DESCRIPTION
**Pull request by emanguy**

### CODE REVIEW

File: src/js/model/blob/blob.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	// Repeatedly read in bytes
+	for (var i = 0; i < blobLength; i++)
+	{
+		try
+		{
+			if (hexadecimal.isEmpty)
+			{
+				throw new RangeError("There were fewer bytes than stated by the length SDNV.");
+			}
+
```
Comment (gti28) [inline src/js/model/blob/blob.js:42]: Only question I have is, should we update the AMPSetup.js to throw an error when we try to call consume when the remaining hex is empty and then handle adding the more detailed message in catch block.  If not feel free to merge

**Review by gti28 — APPROVED** [review]

**Comment by emanguy** [comment]
No, the behavior of AMPHexConsumer is to  have consumeNibbles() do nothing when you hit the end of the hex exactly. It only throws an error if you have hex left and try to consume more than is available. When you hit the exact end you should check for the .isEmpty property. I felt you should handle that case yourself, see my handling in both SDNV and String

File: src/js/model/blob/blob.js (hunk 1 above)
Comment (gti28) [reply src/js/model/blob/blob.js:42]: I agree the handling you have in SDNV and String makes sense,  but I still think we should throw a range exception if we call consume on an empty string.  If we do somehow get in  a case where we call consume on empty string I'd think we'd want the same exception as if we called consume on string that is too short

Comment (emanguy) [reply src/js/model/blob/blob.js:42]: Fair enough. Make an issue for it and I'll throw it in on another branch

**Pull request merged**